File: API/WebApplication1/Models/Medarbejder.cs
namespace WebApplication1.Models
{
    public class Medarbejder : Common
    {
        public required string Navn { get; set; }
        public required string Email { get; set; }
        public required string Password { get; set; }
        public required DateTime ArbejdsdagStart { get; set; }
        public required DateTime ArbejdsdagSlut { get; set; }
        public required int ArbejdstimerOmUgen { get; set; }

        // IKKE TIL PRODUKTION
        public required string RealPassword { get; set; }

        public string? Tlf { get; set; }
        public string? Farve { get; set; }
        List<Modul?> Moduler { get; set; } = new List<Modul?>();
        List<Fravaer?> Fravaer { get; set; } = new List<Fravaer?>();
        public List<Rolle?> Roller { get; set; } = new List<Rolle?>();
    }
    public class SignupDTO
    {
        public required string Navn { get; set; }
        public required string Email { get; set; }
        public required string Password { get; set; }
        public required DateTime ArbejdsdagStart { get; set; }
        public required DateTime ArbejdsdagSlut { get; set; }
        public required int ArbejdstimerOmUgen { get; set; }
        public string? Tlf { get; set; }
        public string? Farve { get; set; }
    }
    public class LoginDTO
    {
        public required string Email { get; set; }
        public required string Password { get; set; }
    }
}

File: Flutter/assignly/lib/classes/objects/medarbejder.dart
import 'dart:convert';

import 'package:assignly/classes/objects/fravaer.dart';
import 'package:assignly/classes/objects/rolle.dart';

class Medarbejder {
  String? id,
      navn, 
      email, 
      password, 
      tlf, 
      farve;
  DateTime? arbejdsdagStart, arbejdsdagSlut;
  int? arbejdstimerOmUgen;
  List<Fravaer?> fravaer = [];
  List<Rolle?> roller = [];

  Medarbejder({
    required this.id,
    required this.navn,
    required this.email,
    required this.password,
    this.tlf,
    this.farve,
    this.arbejdsdagStart,
    this.arbejdsdagSlut,
    this.arbejdstimerOmUgen,
    this.fravaer = const [],
    this.roller = const [],
  });

  static List<Medarbejder> getMedarbejderFromJson(String response) {
    try {
      List<Medarbejder> medarbejdere = [];
      
      final result = jsonDecode(response);
      if (result != null) {
        if (result is List && result.isNotEmpty) {
          for (var element in result) {
            Medarbejder? newMedarbejder = getMedarbejderFromJsonMap(element);
            if (newMedarbejder != null) {
              medarbejdere.add(newMedarbejder);
            }
          }
          return medarbejdere;
        } else if (result is Map<String, dynamic>) {
          Medarbejder? newMedarbejder = getMedarbejderFromJsonMap(result);
          if (newMedarbejder != null) {
            medarbejdere.add(newMedarbejder);
            return medarbejdere;
          } else {
            return [];
          }
        }
      }
    } catch (_) {}

    return [];
  }

  static Medarbejder? getMedarbejderFromJsonMap(Map<String, dynamic> response) {
    try {
      Medarbejder medarbejder = Medarbejder(
        id: response['id'],
        navn: response['navn'],
        email: response['email'],
        password: response['password'],
        tlf: response['tlf'],
        farve: response['farve'],
        arbejdsdagStart: response['arbejdsdagStart'] != null ? DateTime.parse(response['arbejdsdagStart']) : null,
        arbejdsdagSlut: response['arbejdsdagSlut'] != null ? DateTime.parse(response['arbejdsdagSlut']) : null,
        arbejdstimerOmUgen: response['arbejdstimerOmUgen'],
      );

      // Fravær
      var fravaerJson = response['fravaer'];
      if (fravaerJson != null && fravaerJson is List && fravaerJson.isNotEmpty) {
        for (var fravaerElement in fravaerJson) {
          Fravaer? fravaer = Fravaer.getFravaerFromJsonMap(fravaerElement);
          if (fravaer != null) {
            medarbejder.fravaer.add(fravaer);
          }
        }
      }

      // Roller
      var rolleJson = response['roller'];
      if (rolleJson != null && rolleJson is List && rolleJson.isNotEmpty) {
        for (var rolleElement in rolleJson) {
          Rolle? rolle = Rolle.getRolleFromJsonMap(rolleElement);
          if (rolle != null) {
            medarbejder.roller.add(rolle);
          }
        }
      }

      return medarbejder;

    } catch (_) {}

    return null;
  }
}